### ISSUE TITLE
java.lang.NoClassDefFoundError: com/amazon/corretto/crypto/provider/EcUtils$ECInfo

### ISSUE THREAD

**Issue by tginiotis-at-work** [jfrog/jenkins-artifactory-plugin#624]
**Describe the bug**
A remote agent never connects back to a Jenkins server that was restarted. The following stack traces can be seen in the agent's log:

```
INFO: Connecting to REDACTED
Jan 25, 2022 4:19:14 PM hudson.remoting.jnlp.Main$CuiListener status
INFO: Trying protocol: JNLP4-connect
Jan 25, 2022 4:19:14 PM org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Reader run
INFO: Waiting for ProtocolStack to start.
Jan 25, 2022 4:19:15 PM org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Reader run
SEVERE: [JNLP4-connect connection to REDACTED] Reader thread killed by NoClassDefFoundError
java.lang.NoClassDefFoundError: com/amazon/corretto/crypto/provider/EcUtils$ECInfo
        at com.amazon.corretto.crypto.provider.EcGen.<clinit>(EcGen.java:33)
        at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
        at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(Unknown Source)
        at com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider$ACCPService.newInstance(AmazonCorrettoCryptoProvider.java:186)
        at java.base/sun.security.jca.GetInstance.getInstance(Unknown Source)
        at java.base/java.security.KeyPairGenerator.getInstance(Unknown Source)
        at java.base/sun.security.ssl.JsseJce.getKeyPairGenerator(Unknown Source)
        at java.base/sun.security.ssl.ECDHKeyExchange$ECDHEPossession.<init>(Unknown Source)
        at java.base/sun.security.ssl.NamedGroup$ECDHEScheme.createPossession(Unknown Source)
        at java.base/sun.security.ssl.NamedGroup$NamedGroupSpec.createPossession(Unknown Source)
        at java.base/sun.security.ssl.NamedGroup.createPossession(Unknown Source)
        at java.base/sun.security.ssl.SSLKeyExchange$T13KeyAgreement.createPossession(Unknown Source)
        at java.base/sun.security.ssl.SSLKeyExchange.createPossessions(Unknown Source)
        at java.base/sun.security.ssl.KeyShareExtension$CHKeyShareProducer.produce(Unknown Source)
        at java.base/sun.security.ssl.SSLExtension.produce(Unknown Source)
        at java.base/sun.security.ssl.SSLExtensions.reproduce(Unknown Source)
        at java.base/sun.security.ssl.ServerHello$T13HelloRetryRequestConsumer.consume(Unknown Source)
        at java.base/sun.security.ssl.ServerHello$ServerHelloConsumer.onHelloRetryRequest(Unknown Source)
        at java.base/sun.security.ssl.ServerHello$ServerHelloConsumer.consume(Unknown Source)
        at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(Unknown Source)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(Unknown Source)
        at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processRead(SSLEngineFilterLayer.java:380)
        at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecv(SSLEngineFilterLayer.java:117)
        at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecv(ProtocolStack.java:677)
        at org.jenkinsci.remoting.protocol.impl.AckFilterLayer.onRecv(AckFilterLayer.java:255)
        at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecv(ProtocolStack.java:677)
        at org.jenkinsci.remoting.protocol.NetworkLayer.onRead(NetworkLayer.java:136)
        at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer.access$2200(BIONetworkLayer.java:49)
        at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Reader.run(BIONetworkLayer.java:291)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassNotFoundException: com.amazon.corretto.crypto.provider.EcUtils$ECInfo
        at java.base/java.net.URLClassLoader.findClass(Unknown Source)
        at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:215)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 37 more

```

**To Reproduce**
Use Jenkins Artifactory plugin 3.15.2 and `jenkins/inbound-agent` docker image agent as a remote agent. Restart Jenkins & the agent fails to reconnect.

**Expected behavior**
The remote agent reconnects after a few failures.

**Versions**
- Jenkins Artifactory plugin version: 3.15.2
- Jenkins operating system: Linux
- Artifactory Version: Enterprise Plus 7.21.23 rev 72123900

**Comment by yahavi** [comment]
@tginiotis-at-work  
We released Jenkins Artifactory plugin 3.15.3 to target this issue. 
In this release, we [removed](https://github.com/jfrog/build-info/pull/609/files) the AmazonCorrettoCryptoProvider code.

Feel free to upgrade.
We'd appreciate your feedback on that.

**Comment by tginiotis-at-work** [comment]
@yahavi thanks! 
Though #625 and #597 are preventing upgrading past 3.13.2.
